addons/l10n_my_ubl_pint/models/account_edi_xml_pint_my.py

```diff
@@ -0,0 +1,117 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from collections import defaultdict
+
+from odoo import models, _
+
+
+class AccountEdiXmlUBLPINTMY(models.AbstractModel):
+    _inherit = "account.edi.xml.ubl_bis3"
+    _name = 'account.edi.xml.pint_my'
+    _description = "Malaysian implementation of Peppol International (PINT) model for Billing"
+    """
+    * PINT Official documentation: https://docs.peppol.eu/poac/pint/pint/
+    * PINT MY Official documentation: https://docs.peppol.eu/poac/my/pint-my
+    """
+
+    def _export_invoice_filename(self, invoice):
+        # EXTENDS account_edi_ubl_cii
+        return f"{invoice.name.replace('/', '_')}_pint_my.xml"
+
+    def _export_invoice_vals(self, invoice):
+        # EXTENDS account_edi_ubl_cii
+        vals = super()._export_invoice_vals(invoice)
+        vals['vals'].update({
+            # see https://docs.peppol.eu/poac/my/pint-my/bis/#profiles
+            'customization_id': 'urn:peppol:pint:billing-1@my-1',
+            'profile_id': 'urn:peppol:bis:billing',
+        })
+        if invoice.currency_id != invoice.company_id.currency_id:
+            # see https://docs.peppol.eu/poac/my/pint-my/bis/#_tax_in_accounting_currency
+            vals['vals']['tax_currency_code'] = invoice.company_id.currency_id.name  # accounting currency
+        return vals
+
+    def _get_invoice_tax_totals_vals_list(self, invoice, taxes_vals):
+        # EXTENDS account_edi_ubl_cii
+        vals_list = super()._get_invoice_tax_totals_vals_list(invoice, taxes_vals)
+        company_currency = invoice.company_id.currency_id
+        if invoice.currency_id != company_currency:
+            # see https://docs.peppol.eu/poac/my/pint-my/bis/#_tax_in_accounting_currency
+            tax_totals_vals = {
+                'currency': company_currency,
+                'currency_dp': company_currency.decimal_places,
+                'tax_amount': taxes_vals['tax_amount'],
+                'tax_subtotal_vals': [],
+            }
+            for _grouping_key, vals in taxes_vals['tax_details'].items():
+                tax_totals_vals['tax_subtotal_vals'].append({
+                    'currency': company_currency,
+                    'currency_dp': company_currency.decimal_places,
+                    'taxable_amount': vals['base_amount'],
+                    'tax_amount': vals['tax_amount'],
+                    'percent': vals['_tax_category_vals_']['percent'],
+                    'tax_category_vals': vals['_tax_category_vals_'],
+                })
+            vals_list.append(tax_totals_vals)
```

Comment:
Multicurrency is not working currently.

It will raise: "aligned-ibrp-t-08": ("[aligned-ibrp-t-08]-For each different value of tax category rate (ibt-119) where the tax category code (ibt-118) is "Standard rated", the tax category taxable amount (ibt-116) in a tax breakdown (ibg-23) MUST equal the sum of Invoice line net amounts (ibt-131) plus the sum of document level charge amounts (ibt-99) minus the sum of document level allowance amounts (ibt-92) where the tax category code (ibt-151, ibt-102, ibt-95) is "Standard rated" and the tax rate (ibt-152, ibt-103, ibt-96) equals the tax category rate (ibt-119).").

Say you the invoice's currency is in USD, the company is in MYR (rate: 1USD = 2MYR), then what we should obtain is (based on my trials, using the tool of philipp helger):
```xml
<cac:TaxTotal>
  <cbc:TaxAmount currencyID="USD">10.00</cbc:TaxAmount>
    <cac:TaxSubtotal>
      <cbc:TaxableAmount currencyID="USD">100.00</cbc:TaxableAmount>
      <cbc:TaxAmount currencyID="USD">10.00</cbc:TaxAmount>
      <cac:TaxCategory>
        <cbc:ID>T</cbc:ID>
        <cbc:Percent>10.0</cbc:Percent>
        <cac:TaxScheme>
          <cbc:ID>VAT</cbc:ID>
        </cac:TaxScheme>
      </cac:TaxCategory>
    </cac:TaxSubtotal>
  </cac:TaxTotal>
  <cac:TaxTotal>
    <cbc:TaxAmount currencyID="MYR">20.00</cbc:TaxAmount>
  </cac:TaxTotal>
  ```
  So we just need the total tax amount in company currency.
  
  This should do the job:
  ```suggestion
              vals_list.append({
                'currency': company_currency,
                'currency_dp': company_currency.decimal_places,
                'tax_amount': taxes_vals['tax_amount'],
                'tax_subtotal_vals': [],
            })
  ```

Reply:
Nice catch and my bad as above, I didn't test properly. Sorry!